### DISCUSSION TITLE
Avoir l'option commune "class" plutôt que "classes"

### DISCUSSION THREAD

**Discussion by bricedurand** [betagouv/dsfr-view-components#156] [Ideas]
Bonjour à tous,

En commençant à utiliser la gem, j'ai eu besoin de mettre une classe supplémentaire sur un composant.
J'ai donc tenté ma chance comme ça : 
```ruby 
dsfr_badge(status: :info, class: "fr-badge--no-icon")
```

J'ai d'abord cru qu'on ne pouvait pas ajouter de classes, puis vu dans la doc que ça se faisait de cette manière : 
```ruby 
dsfr_badge(status: :info, classes: ["fr-badge--no-icon"])
```

Ça m'a paru bizarre, peu intuitif... je préfère la première syntaxe.

Vous en pensez quoi ?

**Comment by adipasquale** [comment]
hello @bricedurand ! je n’ai pas trop d’avis, si ça te parait plus pertinent je pense que tu peux changer ça. 

mon avis sans trop y réfléchir plus que ça :

`class` étant un keyword ruby ça complique souvent un peu le code - surtout côté librairie à écrire ici mais aussi un peu coté appelant. par exemple j’aime bien la nouvelle syntaxe ruby pour passer les keyword args implicitement par exemple je pourrais faire un truc comme ça : 

```rb
classes = %w[beautiful-label]
classes += %w[bold italic] if thisistrue?
return dsfr_badge(status: :info, classes:)
```

et globalement ils ont l’air assez sérieux chez govuk donc pas hésiter à aller voir dans le code s’il y a une raison ou des commentaires sur ce choix.

Mais dans tous les cas tu es libre de faire ce que tu veux avec le go de Stéphane puisqu’à ma connaissance vous êtes maintenant les deux seuls usagers de cette lib en prod :)

**Reply by bricedurand** [reply]
Je n'avais pas conscience que c'était fait comme ça sur govuk, en effet ils doivent avoir leurs raisons !

Effectivement le fait que class soit un mot clé n'aide pas.

Du coup ça m'a l'air d'introduire trop de complexité pour peu de bénéfices ^^ Autant rester comme ça !
